pkg/utils/k8s/k8s_utils_test.go

```diff
@@ -41,3 +48,160 @@ func TestParseImageHash(t *testing.T) {
 		})
 	}
 }
+
+func TestGetMatchingSecretName(t *testing.T) {
+	notSpecificImageName := "gcr.io/not/specific"
+	specificImageName := "gcr.io/more/specific"
+	dockerconfigjson := credentialprovider.DockerConfigJson{
+		Auths: credentialprovider.DockerConfig(map[string]credentialprovider.DockerConfigEntry{
+			"http://foo.example.com": {
+				Username: "foo",
+				Password: "bar",
+			},
+			"gcr.io": {
+				Username: "gcr",
+				Password: "io",
+			},
+			specificImageName: {
+				Username: "gcr",
+				Password: "io/more/specific",
+			},
+		}),
+	}
+	dockerconfigjsonB, err := json.Marshal(dockerconfigjson)
+	assert.NilError(t, err)
+
+	noMatchDockerconfigjson := credentialprovider.DockerConfigJson{
+		Auths: credentialprovider.DockerConfig(map[string]credentialprovider.DockerConfigEntry{
+			"http://foo.example.com": {
+				Username: "foo",
+				Password: "bar",
+			},
+		}),
+	}
+	noMatchDockerconfigjsonB, err := json.Marshal(noMatchDockerconfigjson)
+	assert.NilError(t, err)
+
+	type args struct {
+		secrets   []*corev1.Secret
+		imageName string
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		{
+			name: "match specific image name",
+			args: args{
+				secrets: []*corev1.Secret{
+					{
+						TypeMeta: metav1.TypeMeta{
+							Kind:       "Secret",
+							APIVersion: "v1",
+						},
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "regcred",
+							Namespace: "default",
+						},
+						Data: map[string][]byte{
+							corev1.DockerConfigJsonKey: dockerconfigjsonB,
+						},
+						Type: corev1.SecretTypeDockerConfigJson,
+					},
+				},
+				imageName: specificImageName,
+			},
+			want: "regcred",
+		},
+		{
+			name: "match not specific image name",
+			args: args{
+				secrets: []*corev1.Secret{
+					{
+						TypeMeta: metav1.TypeMeta{
+							Kind:       "Secret",
+							APIVersion: "v1",
+						},
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "no-match",
```

Comment:
no match?

Reply:
Didn't understand the question.

Reply:
There are 2 secrets here, the test will match the second one named `regcred`